src/dev/mocha/run_mocha_cli.js

```diff
@@ -0,0 +1,75 @@
+import { resolve } from 'path';
+
+import getopts from 'getopts';
+import globby from 'globby';
+
+export function runMochaCli() {
+  const opts = getopts(process.argv.slice(2), {
+    alias: {
+      t: 'timeout',
+    },
+    boolean: [
+      'no-timeouts'
+    ],
+  });
+
+  const runInBand = process.execArgv.some(arg => {
+    switch (arg) {
+      case '--inspect':
+      case '--inspect-brk':
+        return true;
+    }
+  });
+
+  // ensure that mocha exits when test have completed
+  process.argv.push('--exit');
+
+  // ensure that mocha requires the babel-register script
+  process.argv.push('--require', require.resolve('../../babel-register'));
+
+  // set default test timeout
+  if (opts.timeout == null && !opts['no-timeouts']) {
+    if (runInBand) {
+      process.argv.push('--no-timeouts');
+    } else {
+      process.argv.push('--timeout', '10000');
+    }
+  }
+
+  // set default slow timeout
+  if (opts.slow == null) {
+    process.argv.push('--slow', '5000');
+  }
+
+  // set default reporter
+  if (opts.reporter == null) {
+    process.argv.push('--reporter', require.resolve('./server_junit_reporter'));
+  }
+
+  // set default test files
+  if (!opts._.length) {
+    globby.sync([
```

Comment:
This isn't a blocker for this PR or anything, but it's a shame that things like glob paths are buried in the middle of a source file now. The old grunt task setup is flawed in many ways, but the separation of config and task source was nice.

Reply:
yeah, I'm not totally opposed to a similar solution but I personally found the separation unnecessarily abstract and caused configurability that was often buggy, unclear, or didn't seem to make sense. Probably worth having a discussion about though.

Reply:
This seems minor enough that it's just not worth spending cycles on at the moment.